flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/ContextClassLoadingSettingTest.java

```diff
@@ -107,49 +108,87 @@ void shutdown() throws InterruptedException, ExecutionException, TimeoutExceptio
     void testAkkaRpcService_ExecuteRunnableSetsFlinkContextClassLoader()
             throws ExecutionException, InterruptedException {
         final CompletableFuture<ClassLoader> contextClassLoader = new CompletableFuture<>();
-        akkaRpcService.execute(
-                () -> contextClassLoader.complete(Thread.currentThread().getContextClassLoader()));
+        akkaRpcService
+                .getScheduledExecutor()
+                .execute(
+                        () ->
+                                contextClassLoader.complete(
+                                        Thread.currentThread().getContextClassLoader()));
         assertIsFlinkClassLoader(contextClassLoader.get());
     }
 
     @Test
-    void testAkkaRpcService_ExecuteCallableSetsFlinkContextClassLoader()
+    void testAkkaRpcService_ScheduleCallableSetsFlinkContextClassLoader()
             throws ExecutionException, InterruptedException {
-        final CompletableFuture<ClassLoader> contextClassLoader =
-                akkaRpcService.execute(() -> Thread.currentThread().getContextClassLoader());
-        assertIsFlinkClassLoader(contextClassLoader.get());
+        final ClassLoader contextClassLoader =
+                akkaRpcService
+                        .getScheduledExecutor()
+                        .schedule(
+                                () -> Thread.currentThread().getContextClassLoader(),
+                                0,
+                                TimeUnit.MILLISECONDS)
+                        .get();
+        assertIsFlinkClassLoader(contextClassLoader);
```

Comment:
```suggestion
        assertThat(contextClassLoader.get()).isSameAs(pretendFlinkClassLoader);
```
I guess, we can be more restrictive here.